### PR TITLE
catch and handle errors when a user tries to steal another's slack membership

### PR DESCRIPTION
**Pull request by camhashemi**
@leogrim @ryanpbrewster 

this is something I'm running into when QA'ing the backfill scopes link. when a Kifi user is logged into a different account than the one that's linked to the slack user we send the message to, it blows up. This is the LHF such that the user can resolve their own issue, more complicated would be to enforce that you're logged in to the Kifi user we expect before entering this flow.

### CODE REVIEW

**Comment by ryanpbrewster** [comment]
Seems like a good idea to me. :+1:

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackActionFail.scala [hunk 1]
```diff
@@ -31,6 +31,9 @@ object SlackActionFail {
   case class InvalidMembership(userId: Id[User], slackTeamId: SlackTeamId, slackTeamName: SlackTeamName, membership: Option[SlackTeamMembership])
     extends SlackActionFail("invalid_slack_membership", s"User $userId is not a valid member of SlackTeam ${slackTeamName.value} (${slackTeamId.value}): $membership")
 
+  case class MembershipExists(userId: Id[User], ownerId: Id[User], slackTeamId: SlackTeamId, slackTeamName: SlackTeamName, slackUserId: SlackUserId, membership: SlackTeamMembership)
+    extends SlackActionFail("slack_membership_exists", s"$userId is trying to steal existing membership from $ownerId on SlackTeam ${slackTeamName.value} (${slackTeamId.value}): $membership")
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackActionFail.scala:35]: Could we go for `MembershipAlreadyConnected` and be consistent with the team-level error conventions?

**Comment by leogrim** [comment]
Great! :star2:

File: kifi-backend/modules/shoebox/app/com/keepit/slack/SlackActionFail.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/slack/SlackActionFail.scala:35]: :+1: good thinking

**Pull request merged**